src/components/DotNavigation/DotNavigation.spec.tsx

```diff
@@ -0,0 +1,330 @@
+import * as React from 'react';
+import { createUniDriverFactory } from 'wix-ui-test-utils/uni-driver-factory';
+import { isUniEnzymeTestkitExists } from 'wix-ui-test-utils/enzyme';
+import { isUniTestkitExists } from 'wix-ui-test-utils/vanilla';
+import { mount } from 'enzyme';
+import { dotNavigationDriverFactory } from './DotNavigation.driver';
+import { Animation, Theme, DotNavigation } from './DotNavigation';
+import { dotNavigationTestkitFactory } from '../../testkit';
+import { dotNavigationTestkitFactory as enzymeDotNavigationTestkitFactory } from '../../testkit/enzyme';
+
+describe('DotNavigation', () => {
+  const LONG_COMPONENT_LENGTH = 10;
+  const createDriver = createUniDriverFactory(dotNavigationDriverFactory);
+
+  describe('general', () => {
+    it('should render', async () => {
+      const driver = createDriver(<DotNavigation />);
+
+      expect(await driver.exists()).toBe(true);
+    });
+
+    it('should render 5 dots by default', async () => {
+      const driver = createDriver(<DotNavigation />);
+      const dots = driver.getDots();
+
+      expect(await dots.count()).toBe(5);
+    });
+
+    it('should render no dots with negative length', async () => {
+      const driver = createDriver(<DotNavigation length={-5} />);
+      const dots = driver.getDots();
+
+      expect(await dots.get(0).exists()).toBe(false);
+    });
+
+    it('should render no dots with 0 length', async () => {
+      const driver = createDriver(<DotNavigation length={0} />);
+      const dots = driver.getDots();
+
+      expect(await dots.get(0).exists()).toBe(false);
+    });
+
+    it('should render 4 dots with length equal to 4', async () => {
+      const driver = createDriver(<DotNavigation length={4} />);
+      const dots = driver.getDots();
+
+      expect(await dots.count()).toBe(4);
+    });
+
+    it('should render 9 dots with length greater than 5', async () => {
+      const driver = createDriver(<DotNavigation length={6} />);
```

Comment:
I feel like i'm missing something here.
The length passed to the component is `6` but the result is `9`?

Reply:
That's true. If length > 5, 9 dots are rendered:
3 middle dots,
2 small dots for moving back and forward,
2 tiny dots for going to start and to end,
2 additional tiny dots at the beginning and at the end for correct animation.

Reply:
I see, so maybe we should have a different name for this.
The driver's methods will also be used by consumers (via the testkit), it might confuse them.
Maybe we should have a distinction between `getVisibleDots` and `getRenderedDots`.
I'm not sure the naming is good, but something like that.

Reply:
As component logic changed, this test is no longer valid.